src/dbnode/storage/shard.go

```diff
@@ -2660,74 +2661,217 @@ func (s *dbShard) Repair(
 
 func (s *dbShard) AggregateTiles(
 	ctx context.Context,
-	reader fs.DataFileSetReader,
 	sourceNsID ident.ID,
-	sourceBlockStart time.Time,
-	sourceShard databaseShard,
+	sourceShardID uint32,
+	blockReaders []fs.DataFileSetReader,
+	sourceBlockVolumes []shardBlockVolume,
 	opts AggregateTilesOptions,
-	wOpts series.WriteOptions,
+	targetSchemaDescr namespace.SchemaDescr,
 ) (int64, error) {
-	latestSourceVolume, err := sourceShard.latestVolume(sourceBlockStart)
+	blockReadersToClose := make([]fs.DataFileSetReader, 0, len(blockReaders))
```

Comment:
I would instead pass a time range, i.e. start and volume, and then a scoped interface that can return the LatestVolume for a block start?

i.e. 
```go
type databaseShard interface {
  AggregateTiles(
	ctx context.Context,
	start, end time.Time,
	sourceNsID ident.ID,
	sourceShardID uint32,
	sourceShardVolumes sourceShardVolumes,
	blockReaders []fs.DataFileSetReader,
	opts AggregateTilesOptions,
	targetSchemaDescr namespace.SchemaDescr,
  ) (int64, error)
}

type sourceShardVolumes interface {
  LatestVolume(blockStart time.Time) (int, error)
}
```

That way you query for the latest volume within this scoped call and we don't have to pass two arrays of the same length and index into each other here. Also you can pass the sourceShard directly as sourceShardVolumes since it will already fulfill the interface.